hoomd/hpmc/UpdaterBoxMC.h

```diff
@@ -214,65 +280,33 @@ class UpdaterBoxMC : public Updater
 
-        //! Get volume change parameter
-        const Scalar get_volume_delta() const
-            {
-            return m_Volume_delta;
-            }
-
-        //! Get delta_lnV
-        const Scalar get_ln_volume_delta() const
-            {
-            return m_lnVolume_delta;
-            }
-
-
-        //! Get aspect ratio trial parameter
-        const Scalar get_aspect_delta() const
-            {
-            return m_Aspect_delta;
-            }
-
-        //! Get box length trial parameters
-        pybind11::tuple get_length_delta() const
-            {
-            return pybind11::make_tuple(m_Length_delta[0], m_Length_delta[1], m_Length_delta[2]);
-            }
-
-        //! Get box shear trial parameters
-        pybind11::tuple get_shear_delta() const
-            {
-            return pybind11::make_tuple(m_Shear_delta[0], m_Shear_delta[1], m_Shear_delta[2]);
-            }
-
-
     private:
         std::shared_ptr<IntegratorHPMC> m_mc;     //!< HPMC integrator object
-        std::shared_ptr<Variant> m_P;             //!< Reduced pressure in isobaric ensembles
-        Scalar m_frequency;                         //!< Frequency of BoxMC moves versus HPMC integrator moves
+        std::shared_ptr<Variant> m_beta_P;        //!< Reduced pressure in isobaric ensembles
 
-        Scalar m_Volume_delta;                      //!< Amount by which to change parameter during box-change
-        float m_Volume_weight;                     //!< relative weight of volume moves
-        Scalar m_lnVolume_delta;                      //!< Amount by which to change parameter during box-change
-        float m_lnVolume_weight;                   //!< relative weight of volume moves
-        Scalar m_Volume_A1;                         //!< Ratio of Lx to Ly to use in isotropic volume changes
-        Scalar m_Volume_A2;                         //!< Ratio of Lx to Lz to use in isotropic volume changes
+        Scalar m_volume_delta;                    //!< Amount by which to change volume during box-change
+        float m_volume_weight;                    //!< relative weight of volume moves
+        Scalar m_ln_volume_delta;                 //!< Amount by which to log volume parameter during box-change
+        float m_ln_volume_weight;                 //!< relative weight of log volume moves
+        std::string m_volume_mode;                //!< volume moves mode: standard or logarithmic
+        Scalar m_volume_A1;                       //!< Ratio of Lx to Ly to use in isotropic volume changes
+        Scalar m_volume_A2;                       //!< Ratio of Lx to Lz to use in isotropic volume changes
 
-        Scalar m_Length_delta[3];                   //!< Max length change in each dimension
-        float m_Length_weight;                     //!< relative weight of length change moves
+        Scalar m_length_delta[3];                  //!< Max length change in each dimension
```

Comment:
Is there a reason we are not using `Scalar3` @joaander?

Reply:
The original author of the code wrote it this way. I think either way is fine, but at this point changing it to `Scalar3` also requires updating the for loops over dimensions in the implementation.

Reply:
That is fine then. I was wondering if there was a non-historic reason for the choice. This does not need to be addressed here. It would be nice to be consistent though throughout the code base on this eventually, but honestly that is pretty low priority.